Add smoke test tag for critical login scenarios

salopower opened the pull request:
Marked essential login tests with @smoke tag to enable targeted test runs:
- Login with valid credentials
- Login page UI validation
- Error handling for invalid inputs

Run smoke tests with: `npx playwright test --grep=@smoke`

salopower committed:
Add smoke test tag for critical login scenarios

Marked essential login tests with @smoke tag to enable targeted test runs:
- Login with valid credentials
- Login page UI validation
- Error handling for invalid inputs

Run smoke tests with: `npx playwright test --grep=@smoke`

vlevkoniuk commented on home_works/home_work_19/src/pages/orange-hrm-login.page.ts:
```diff
@@ -0,0 +1,107 @@
+    private readonly loginUrl = 'https://opensource-demo.orangehrmlive.com/web/index.php/auth/login';
+    private readonly dashboardUrl = 'https://opensource-demo.orangehrmlive.com/web/index.php/dashboard/index';
+
+    // ========== Constructor ==========
```
а нам коменти потрібні чи забув видалити?

salopower replied:
Коменти залишив, бо як для мене так простіше читати й відокремлювати для себе шматки коду. Можу прибрати якщо потрібно)

vlevkoniuk commented on home_works/home_work_19/tests/specs/login/negative-login.spec.ts:
```diff
@@ -0,0 +1,39 @@
+        await loginPage.goto();
+    });
+
+    test('should show error with wrong password', async () => {
+        await loginPage.login(
+            process.env.TEST_USERNAME as string,
+            'wrongpassword'
+        );
+        await loginPage.verifyFailedLogin();
+        await loginPage.verifyFailedLoginIcon();
+        await loginPage.verifyFailedLoginWithMessage('Invalid credentials');
+    });
+
+    test('should show error with wrong username', async () => {
+        await loginPage.login(
+            'WrongUser',
+            process.env.TEST_PASSWORD as string
+        );
+        await loginPage.verifyFailedLogin();
+        await loginPage.verifyFailedLoginWithMessage('Invalid credentials');
+    });
+
+    test('should show error with special characters in username', async () => {
+        await loginPage.login(
+            process.env.TEST_USERNAME as string + '!@#',
+            process.env.TEST_PASSWORD as string
+        );
+        await loginPage.verifyFailedLogin();
+        await loginPage.verifyFailedLoginWithMessage('Invalid credentials');
+    });
```
параметризуй тест
https://playwright.dev/docs/test-parameterize

salopower replied:
Прийшлося завантажити .env на початку тесту, бо в параметриованих тестах чомусь не отримувалися логін та паролть звідти

vlevkoniuk commented on home_works/home_work_19/tests/specs/login/positive-login.spec.ts:
```diff
@@ -0,0 +1,34 @@
+    });
+
+    test('should persist session after login', async ({ context }) => {
+        await loginPage.login(
+            process.env.TEST_USERNAME as string,
+            process.env.TEST_PASSWORD as string
+        );
+        await loginPage.verifySuccessfulLogin();
```
ці рядки повністю дублюють перший тест

vlevkoniuk commented on home_works/home_work_19/tests/specs/login/ui-elements.spec.ts:
```diff
@@ -0,0 +1,42 @@
+test.describe('Login Page UI Elements @smoke', () => {
+    let loginPage: OrangeHRMLoginPage;
+
+    test.beforeEach(async ({ page, context }) => {
+        loginPage = new OrangeHRMLoginPage(page, context);
+        await loginPage.goto();
+    });
+
+    test('should display company logo', async () => {
+        await expect(loginPage.logo).toBeVisible();
+    });
+
+    test('should display login title with correct text', async () => {
+        await loginPage.verifyLoginTitleVisible();
+        await loginPage.verifyLoginTitleText('Login');
+    });
+
+    test('should display username field with correct label and placeholder', async () => {
+        await expect(loginPage.userNameInput).toBeVisible();
+        await expect(loginPage.userNameInputTitle).toHaveText('Username');
+        await expect(loginPage.userNameInput).toHaveAttribute('placeholder', 'Username');
+    });
+
+    test('should display password field with correct label and placeholder', async () => {
+        await expect(loginPage.passwordInput).toBeVisible();
+        await expect(loginPage.passwordInputTitle).toHaveText('Password');
+        await expect(loginPage.passwordInput).toHaveAttribute('placeholder', 'Password');
+    });
+
+    test('should display login button', async () => {
+        await expect(loginPage.loginButton).toBeVisible();
+        await expect(loginPage.loginButton).toHaveText('Login');
+    });
+
+    test('should display forgot password link', async () => {
+        await expect(loginPage.forgotPasswordLink).toBeVisible();
+        await expect(loginPage.forgotPasswordLink).toHaveText('Forgot your password?');
+    });
```
тут можна перевірити 1 тестом, щоб не завантажувати одну і ту ж сторінку по 10 разів
тільки оце, що зараз в тесті можна описати методом в ПОМ і просто в одному тесті використати методи

vlevkoniuk reviewed (changes requested):
непогано.
накдив коментів, що треба покращити

salopower committed:
refactor: modular login page checks

- cleaning up comments
- parameterization of tests
- creating new methods for checking ui elements

vlevkoniuk reviewed (approved):
looks good!
